modules/integrations/arrow-integrations-retrofit-adapter/src/main/kotlin/arrow.integrations.retrofit.adapter/CallK.kt

```diff
@@ -0,0 +1,33 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.Kind
+import arrow.effects.typeclasses.Async
+import arrow.effects.typeclasses.MonadDefer
+import arrow.typeclasses.MonadError
+import retrofit2.Call
+import retrofit2.HttpException
+import retrofit2.Response
+
+data class CallK<R>(val call: Call<R>) {
+  fun <F> async(async: Async<F>): Kind<F, R> =
```

Comment:
I'd replace this version with the one that uses `Kind<F, Response<R>>` as the return type. I wouldn't impose the semantics of `handleResponse` because you don't give users the chance to inspect other fields of the response.